### PR TITLE
Patches for Mounted Archer, Bow Expert, Crossbow Cavalry, and Crossbow Expert

### PR DESCRIPTION
**Pull request by ajseavey**
Addresses #10 

Heavily based on my crossbow cavalry perk, mostly just a change in condition and the field being changed. Currently the only difference in the item usage between bows and long bows is whether they are able to be used on mounts. This just will conditionally overwrite the item usage of a long bow with that of a bow based on the Agent's perks.

### CODE REVIEW

File: src/CommunityPatch/Patches/AgentWeaponEquippedPatch.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      if (Applied) return false;
+
+      var patchInfo = Harmony.GetPatchInfo(TargetMethodInfo);
+      if (AlreadyPatchedByOthers(patchInfo))
+        return false;
+
+      return _IsApplicable(game);
+    }
+
+    private static void PostFix(ref Agent __instance,
```
Comment (Tyler-IN) [inline src/CommunityPatch/Patches/AgentWeaponEquippedPatch.cs:38]: Given the Patch call above now applies this as the prefix parameter, does this need to be renamed to Prefix?

Also, postfix (proper; suffix) and prefix are full words, so Pascal naming convention would be to just capitalize the P.

File: src/CommunityPatch/ApplicationVersionComparer.cs [hunk 2]
```diff
@@ -25,8 +25,10 @@ public class ApplicationVersionComparer : IComparer<ApplicationVersion>, IEquali
     public bool Equals(ApplicationVersion x, ApplicationVersion y)
       => Compare(x, y) == 0;
 
+    public bool GreaterThan(ApplicationVersion x, ApplicationVersion y) => Compare(x, y) > 0;
+
     public int GetHashCode(ApplicationVersion obj)
-      => ((int) _typeComparer.GetHashCode(obj.ApplicationVersionType) << 28) | (obj.Major << 23) | (obj.Minor << 12) | obj.Revision;
+    => ((int) _typeComparer.GetHashCode(obj.ApplicationVersionType) << 28) | (obj.Major << 23) | (obj.Minor << 12) | obj.Revision;
 
   }
```
Comment (Tyler-IN) [inline src/CommunityPatch/ApplicationVersionComparer.cs:33]: Can you revert these minor formatting changes, they might be accidentally committed?
They're out of scope for the PR.

**Comment by Tyler-IN** [comment]
Title of PR needs to be adjusted to account for the merge for record keeping.

Other issues need to be tagged.

File: src/CommunityPatch/Patches/AgentWeaponEquippedPatch.cs (hunk 1 above)
Comment (ajseavey) [reply src/CommunityPatch/Patches/AgentWeaponEquippedPatch.cs]: Fixed

File: src/CommunityPatch/ApplicationVersionComparer.cs (hunk 2 above)
Comment (ajseavey) [reply src/CommunityPatch/ApplicationVersionComparer.cs:31]: Fixed

Comment (Tyler-IN) [reply src/CommunityPatch/ApplicationVersionComparer.cs:31]: This shouldn't show up in the PR as modified...

**Pull request merged**